src/ui/Analytics/Analytics.ts

```diff
@@ -158,7 +162,69 @@ export class Analytics extends Component {
      * Default value is `undefined`, and the value of this parameter will fallback to the organization used for the
      * search endpoint.
      */
-    organization: ComponentOptions.buildStringOption()
+    organization: ComponentOptions.buildStringOption(),
+
+    /**
+     * Whether to push Coveo usage analytics events to the Google Tag Manager [data layer](https://developers.google.com/tag-manager/devguide#datalayer).
+     *
+     * **Notes:**
+     * - Set this option to `false` to ensure that the component does _not_ populate the data layer, even if a GTM container snippet is present in the page.
+     * - Set this option to `true` to ensure that the component populates the data layer even if a GTM container snippet cannot be found in the page.
+     *
+     * See also [`gtmDataLayerName`]{@link Analytics.options.gtmDataLayerName}.
+     *
+     * By default, if a standard GTM container snippet is found in the page, the component automatically attempts to push to its initialized data layer (see [Google Tag Manager Quick Start Guide](https://developers.google.com/tag-manager/quickstart)).
+     */
+    pushToGtmDataLayer: ComponentOptions.buildBooleanOption({
```

Comment:
Essentially, the default behavior is:

- If no GTM container script is found in the page, do nothing.
- If a GTM container snippet is found, set `pushToGtmDataLayer` to `true`, and attempt to extract the data layer name from the GTM container snippet and set `gtmDataLayerName` to this value (set `gtmDataLayerName` to `"dataLayer"` if no data layer name can be extracted from the snippet). If the data layer object is undefined, don't register the handler (to avoid repeatedly trying to push to a non-existing object).

Reply:
Is the idea that someone who has a GTM script in their page can use this option to tell the search-ui framework not to push to it?

If so, setting pushToGtmDataLayer to `true` automatically when the script is there does not make sense to me.

Regarding extracting the data-layer name, would it be simpler to ask the user to configure it? Maybe this option could become `gtmDataLayerName`? If it is specified, we push to it?

What do you think of having a slack call with share screen to discuss?